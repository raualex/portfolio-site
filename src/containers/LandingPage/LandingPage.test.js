import React from 'react';
import { LandingPage, mapStateToProps, mapDispatchToProps } from './';
import { shallow } from 'enzyme';

describe('LandingPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LandingPage selectedNav={''} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});

describe('mapStateToProps function', () => {
  it('should return an object with selected Navigation object', () => {
    const mockState = {
      selectedNav: 'Landing Page',
      somethingElse: 'Rock Onnn'
    }
    const expected = {
      selectedNav: 'Landing Page'
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps function', () => {
  const mockDispatch = jest.fn()
  let mappedProps;

  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });

  it('should call dispatch when setSelectedProj is called', () => {
    mappedProps.setSelectedProj()
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when setSelectedNav is called', () => {
    mappedProps.setSelectedNav()
    expect(mockDispatch).toHaveBeenCalled()
  });
});