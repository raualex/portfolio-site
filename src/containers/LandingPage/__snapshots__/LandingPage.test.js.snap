// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    selectedNav=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome to my Portfolio Site!
        </h1>,
        <p>
          Feel free to explore and check out my projects
        </p>,
        <a
          className="portfolio-github-link"
          href="https://github.com/raualex/portfolio-site"
          rel="noopener noreferrer"
          target="_blank"
        >
          Click here to see the code for this site
        </a>,
        <p>
          This site was built using:
        </p>,
        <div
          className="landing-logo-container"
        >
          <img
            alt="react logo"
            className="landing-logos-react"
            src="react-logo.png"
          />
          <img
            alt="react router logo"
            className="landing-logos-react-router"
            src="react-router-logo.png"
          />
          <img
            alt="redux logo"
            className="landing-logos-redux"
            src="redux-logo.png"
          />
          <img
            alt="css logo"
            className="landing-logos-css"
            src="css3_logo.svg"
          />
        </div>,
        <p>
          This site was tested using:
        </p>,
        <div
          className="testing-logo-container"
        >
          <img
            alt="enzyme and jest logos"
            className="testing-logos"
            src="enzyme-jest.png"
          />
          <img
            alt="travis ci logo"
            className="testing-logos"
            src="travisCI-logo.png"
          />
        </div>,
      ],
      "className": "lp-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to my Portfolio Site!",
        },
        "ref": null,
        "rendered": "Welcome to my Portfolio Site!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Feel free to explore and check out my projects",
        },
        "ref": null,
        "rendered": "Feel free to explore and check out my projects",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click here to see the code for this site",
          "className": "portfolio-github-link",
          "href": "https://github.com/raualex/portfolio-site",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Click here to see the code for this site",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This site was built using:",
        },
        "ref": null,
        "rendered": "This site was built using:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="react logo"
              className="landing-logos-react"
              src="react-logo.png"
            />,
            <img
              alt="react router logo"
              className="landing-logos-react-router"
              src="react-router-logo.png"
            />,
            <img
              alt="redux logo"
              className="landing-logos-redux"
              src="redux-logo.png"
            />,
            <img
              alt="css logo"
              className="landing-logos-css"
              src="css3_logo.svg"
            />,
          ],
          "className": "landing-logo-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "react logo",
              "className": "landing-logos-react",
              "src": "react-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "react router logo",
              "className": "landing-logos-react-router",
              "src": "react-router-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "redux logo",
              "className": "landing-logos-redux",
              "src": "redux-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "css logo",
              "className": "landing-logos-css",
              "src": "css3_logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This site was tested using:",
        },
        "ref": null,
        "rendered": "This site was tested using:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="enzyme and jest logos"
              className="testing-logos"
              src="enzyme-jest.png"
            />,
            <img
              alt="travis ci logo"
              className="testing-logos"
              src="travisCI-logo.png"
            />,
          ],
          "className": "testing-logo-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "enzyme and jest logos",
              "className": "testing-logos",
              "src": "enzyme-jest.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "travis ci logo",
              "className": "testing-logos",
              "src": "travisCI-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to my Portfolio Site!
          </h1>,
          <p>
            Feel free to explore and check out my projects
          </p>,
          <a
            className="portfolio-github-link"
            href="https://github.com/raualex/portfolio-site"
            rel="noopener noreferrer"
            target="_blank"
          >
            Click here to see the code for this site
          </a>,
          <p>
            This site was built using:
          </p>,
          <div
            className="landing-logo-container"
          >
            <img
              alt="react logo"
              className="landing-logos-react"
              src="react-logo.png"
            />
            <img
              alt="react router logo"
              className="landing-logos-react-router"
              src="react-router-logo.png"
            />
            <img
              alt="redux logo"
              className="landing-logos-redux"
              src="redux-logo.png"
            />
            <img
              alt="css logo"
              className="landing-logos-css"
              src="css3_logo.svg"
            />
          </div>,
          <p>
            This site was tested using:
          </p>,
          <div
            className="testing-logo-container"
          >
            <img
              alt="enzyme and jest logos"
              className="testing-logos"
              src="enzyme-jest.png"
            />
            <img
              alt="travis ci logo"
              className="testing-logos"
              src="travisCI-logo.png"
            />
          </div>,
        ],
        "className": "lp-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to my Portfolio Site!",
          },
          "ref": null,
          "rendered": "Welcome to my Portfolio Site!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Feel free to explore and check out my projects",
          },
          "ref": null,
          "rendered": "Feel free to explore and check out my projects",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click here to see the code for this site",
            "className": "portfolio-github-link",
            "href": "https://github.com/raualex/portfolio-site",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Click here to see the code for this site",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This site was built using:",
          },
          "ref": null,
          "rendered": "This site was built using:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="react logo"
                className="landing-logos-react"
                src="react-logo.png"
              />,
              <img
                alt="react router logo"
                className="landing-logos-react-router"
                src="react-router-logo.png"
              />,
              <img
                alt="redux logo"
                className="landing-logos-redux"
                src="redux-logo.png"
              />,
              <img
                alt="css logo"
                className="landing-logos-css"
                src="css3_logo.svg"
              />,
            ],
            "className": "landing-logo-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "react logo",
                "className": "landing-logos-react",
                "src": "react-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "react router logo",
                "className": "landing-logos-react-router",
                "src": "react-router-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "redux logo",
                "className": "landing-logos-redux",
                "src": "redux-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "css logo",
                "className": "landing-logos-css",
                "src": "css3_logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This site was tested using:",
          },
          "ref": null,
          "rendered": "This site was tested using:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="enzyme and jest logos"
                className="testing-logos"
                src="enzyme-jest.png"
              />,
              <img
                alt="travis ci logo"
                className="testing-logos"
                src="travisCI-logo.png"
              />,
            ],
            "className": "testing-logo-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "enzyme and jest logos",
                "className": "testing-logos",
                "src": "enzyme-jest.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "travis ci logo",
                "className": "testing-logos",
                "src": "travisCI-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
